### PR TITLE
chore: url → Eliminate obsolete "url" library from the cli subpackage

### PR DESCRIPTION
**Pull request by JJ**
- Contributes to #19031. Since `url` is obsolete, there's a plug-in substitute in `URL`, which is core. This is just a PoC that changes 2 of the half dozen instances.

### User facing changelog

Phase out external `url` for core `url`. It was only in `cli`, with the core `url` used elsewhere, which made it a bit strange.

### Additional details

To avoid introducing a deprecated dependence, `querystring`

### How has the user experience changed?

When all the substitution have been made, `url` will be eliminated from deps and there will be no more messages complaining about querystring

### CODE REVIEW

**Review by jennifer-shehane — DISMISSED** [review]

**Review by emilyrohrbough — CHANGES_REQUESTED** [review]
Please make this change in `cli/lib/tasks/install.js` as well and remove the `url` dependency from the package.json.

File: cli/lib/tasks/download.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ const arch = require('arch')
 const la = require('lazy-ass')
 const is = require('check-more-types')
 const os = require('os')
-const Url = require('url')
+const Url = require('URL')
```
Comment (emilyrohrbough) [inline cli/lib/tasks/download.js:5]: The [node documentation](https://nodejs.org/dist/latest-v16.x/docs/api/url.html#url) says this import should be:
```suggestion
const url = require('url');
```

**Comment by JJ** [comment]
> Please make this change in `cli/lib/tasks/install.js` as well and remove the `url` dependency from the package.json.

There are quite a few more:

```
cli/lib/tasks/install.js:const url = require('url')
packages/https-proxy/lib/server.js:const url = require('url')
packages/https-proxy/lib/util/parse.js:const url = require('url')
packages/server/lib/file_server.js:const url = require('url')
packages/server/lib/gui/auth.js:const url = require('url')
packages/server/lib/request.js:const url = require('url')
packages/server/lib/util/origin.js:const url = require('url')
packages/server/test/integration/http_requests_spec.js:const url = require('url')
system-tests/projects/e2e/cypress/integration/node_builtins_spec.js:  expect(require('url')).to.be.an('object').and.have.property('resolve')
```

Can change them all, if that's OK with you. I just did the initial PR to probe if this was really worth the while (and also if it triggered some more errors somewhere else)

**Comment by emilyrohrbough** [comment]
@JJ  to correctly close the issue you linked, we will need to update the `url` dependency in the `packages` modules. This does lead into this issue #19407 and we'd likely want to evaluate how we can align on these before doing a direct replacement of the url module with node's url implementation.

The contents within the packages directly are bundled into the Cypress binary which is executed when Cypress is ran. These dependencies would not be observed within the dependency tree produced by `npm list url`. This is observed because it is a dependency of the cypress cli package (`/cli`).

To at least clean up the deprecated npm warnings on install, we'll want to remove the `url` module dependency from the package.json.

File: cli/lib/tasks/download.js (hunk 1 above)
Comment (JJ) [reply cli/lib/tasks/download.js]: Done in a161000

**Comment by JJ** [comment]
> @JJ to correctly close the issue you linked, we will need to update the `url` dependency in the `packages` modules. This does lead into this issue #19407 and we'd likely want to evaluate how we can align on these before doing a direct replacement of the url module with node's url implementation.
> 
> The contents within the packages directly are bundled into the Cypress binary which is executed when Cypress is ran. These dependencies would not be observed within the dependency tree produced by `npm list url`. This is observed because it is a dependency of the cypress cli package (`/cli`).
> 
> To at least clean up the deprecated npm warnings on install, we'll want to remove the `url` module dependency from the package.json.

Right. Done in the last commit.

**Comment by JJ** [comment]
> @JJ to correctly close the issue you linked, we will need to update the `url` dependency in the `packages` modules. This does lead into this issue #19407 and we'd likely want to evaluate how we can align on these before doing a direct replacement of the url module with node's url implementation.

In principle, this PR touches that tangentially, but I'll wait for the test results to know what's what. But it would probably be worth the while to substitute qs by the core query-string to make it consistent. That will eliminate a dependency in request.

**Comment by emilyrohrbough** [comment]
@JJ Can you take a look at the failing unit tests?

**Comment by JJ** [comment]
I guess I'll have to. Looks like it's trying to download something and it can't, so it might be a flapper or might [something wrong](https://app.circleci.com/pipelines/github/cypress-io/cypress/29656/workflows/9d11feac-e7b5-44fb-87ef-a30c11157dee/jobs/1143544). Definitely worth a look, I guess.

File: cli/lib/tasks/download.js [hunk 2]
```diff
@@ -346,7 +346,7 @@ const start = (opts) => {
     return getCA()
   })
   .then((ca) => {
-    return downloadFromUrl({ url, downloadDestination, progress, ca, version,
+    return downloadFromUrl({ versionUrl, downloadDestination, progress, ca, version,
```
Comment (emilyrohrbough) [inline cli/lib/tasks/download.js:349]:
```suggestion
    return downloadFromUrl({ url: versionUrl, downloadDestination, progress, ca, version,
```
@JJ This should fix the unit test failures

**Review by mjhenkes — APPROVED** [review]

**Review by jennifer-shehane — APPROVED** [review]

**Review by emilyrohrbough — APPROVED** [review]

**Pull request merged**